src/trip/TripContext.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { createContext, useState } from 'react'
+
+import { getCallApi, postCallApi, putCallApi, deleteCallApi } from '../utils/api/fetchMiddleware'
+import { USERS_ENDPOINT, TRIPS_ENDPOINT } from './constants'
+import { useEffect } from 'react';
+
+export const TripContext = createContext();
+
+const TripContextProvider = (props) => {
+    const [users, setUsers] = useState()
+    const [trips, setTrips] = useState()
+
+
+    const createTrip = async (trip) => { 
+        const response = await postCallApi(TRIPS_ENDPOINT, trip)
+        if (response.ok) {
+            const jsonData = await response.json()
+            console.log(jsonData)
+            setTrips([...trips, jsonData.data])
+        }
+    }
+
+    const updateTrip = async (id, trip) => {
+        const response = await putCallApi(`${TRIPS_ENDPOINT}/${id}`, trip)
+        if (response.ok) {
+            const jsonData = await response.json()
+            const copy = [...trips]
+            const index = trips.findIndex(trip => trip._id === jsonData.data._id)
+            copy[index] = jsonData.data
+            setTrips([...copy])
+        }
+    }
+
+    const deleteTrip = async (id) => {
+        const response = await deleteCallApi(`${TRIPS_ENDPOINT}/${id}`)
+        if (response.ok) {
+            const jsonData = await response.json()
+            const copy = [...trips]
+            const index = trips.findIndex(trip => trip._id === jsonData.data._id)
+            copy.splice(index, 1)
+            setTrips([...copy])
```

Comment:
It's because it is not recommended to mutate state directly, the updates would not be tracked properly, so the recommended way is to copy state, perform the update on the copy and then pass the updated copy into the setter.

Reply:
Yeah I get that, but then why do you copy again the updated copy by passing `[...copy]` ? Actually copy is a very poor name, use `newTrips` or something with a similar meaning

Reply:
refactored